lms/djangoapps/instructor_task/api_helper.py

```diff
@@ -75,6 +95,36 @@ def _reserve_task(course_id, task_type, task_key, task_input, requester):
     return InstructorTask.create(course_id, task_type, task_key, task_input, requester)
 
 
+def generate_already_running_error_message(task_type):
+    """
+
+    Returns already running error message for given task type.
+    """
+
+    message = ''
+    already_running_message_template = "The {} report is being created. " \
+                                       "To view the status of the report, see Pending Tasks below. " \
+                                       "You will be able to download the report when it is complete."
+    report_types = {
```

Comment:
`report_type` is constructed to separate out common part of successful message string.
In `instructor_task/api.py` we construct `task_type` not `report_type`. `task_type` is used as field of task's db entry. If you mean to pass `task_type` in `submit_*` methods of `instructor_task/api.py`, I think we can not generalize that because `report_type` is used in smaller subset of `instructor_task/api.py` calls relatively to `task_type`. `task_type` is used in all `submit_*` methods of `instructor_task/api.py`.
Happy to discuss it more.